test/Common/HttpApiHandlerTests.cs

```diff
@@ -62,6 +62,45 @@ public async Task WaitForCompletionOrCreateCheckStatusResponseAsync_Throws_Excep
             Assert.Equal($"Total timeout 0 should be bigger than retry timeout 100", ex.Message);
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void OutOfProcEndpoints_UpdateWithUpdatedWebhookUri()
+        {
+            var httpApiHandler = new HttpApiHandler(GetTestExtension(), null);
+            var webhookProvider = new ChangingWebhookProvider() { WebhookUri = new Uri(TestConstants.NotificationUrl) };
+            httpApiHandler.RegisterWebhookProvider(() => webhookProvider.WebhookUri);
+
+            AssertApisUsingCorrectWebhookUri(httpApiHandler, TestConstants.NotificationUrlBase);
+
+            string newWebhookUri = TestConstants.NotificationUrl.Replace("localhost:7071", "localhost:5050");
+            string newBaseUri = TestConstants.NotificationUrlBase.Replace("localhost:7071", "localhost:5050");
+            webhookProvider.WebhookUri = new Uri(newWebhookUri);
+
+            AssertApisUsingCorrectWebhookUri(httpApiHandler, newBaseUri);
+        }
+
+        // Validate the expected uris are used for CreateHttpManagementPayload(), GetBaseUrl(), and GetInstanceCreationLinks()
+        private static void AssertApisUsingCorrectWebhookUri(HttpApiHandler httpApiHandler, string expectedBaseUri)
+        {
+            HttpManagementPayload managementPayload = httpApiHandler.CreateHttpManagementPayload(
+                 TestConstants.InstanceId,
+                 null,
+                 null);
+
+            Assert.StartsWith(expectedBaseUri, managementPayload.StatusQueryGetUri);
+            Assert.StartsWith(expectedBaseUri, managementPayload.SendEventPostUri);
+            Assert.StartsWith(expectedBaseUri, managementPayload.PurgeHistoryDeleteUri);
+            Assert.StartsWith(expectedBaseUri, managementPayload.RestartPostUri);
+            Assert.StartsWith(expectedBaseUri, managementPayload.TerminatePostUri);
+
+            string baseUri = httpApiHandler.GetBaseUrl();
+            Assert.Equal(expectedBaseUri, baseUri);
+
+            HttpCreationPayload creationPayload = httpApiHandler.GetInstanceCreationLinks();
+            Assert.StartsWith(expectedBaseUri, creationPayload.CreateNewInstancePostUri);
+            Assert.StartsWith(expectedBaseUri, creationPayload.CreateAndWaitOnNewInstancePostUri);
+        }
+

```

Comment:
Is this actually testing the behavior introduced in this PR or is this just a missing test? I'm asking because I don't see any URI updating in this test.

Reply:
This is a helper method for the test that updates the URI, immediately above this method.